include/Surelog/Common/FileSystem.h

```diff
@@ -0,0 +1,70 @@
+/*
+ Copyright 2022 chipsalliance
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+/*
+ * File:   FileSystem.cpp
+ * Author: hs
+ *
+ * Created on June 1, 2022, 3:00 AM
+ */
+
+#ifndef SURELOG_FILESYSTEM_H
+#define SURELOG_FILESYSTEM_H
+#pragma once
+
+#include <cstdint>
+#include <filesystem>
+#include <string_view>
+
+namespace SURELOG {
+class PathId;
+class SymbolId;
+class SymbolTable;
+
+class FileSystem {
+ public:
+  static FileSystem *getInstance();
+  static FileSystem *setInstance(FileSystem *const fileSystem);
+
+ public:
+  virtual PathId toPathId(const std::filesystem::path &path,
```

Comment:
For something called `FileSystem` it does not contain anything that a file system would provide, like creating/removing/open/closing files.

Maybe the naming is off ? Are there plans to expand this ?